rqd/rqd/rqcore.py

```diff
@@ -99,8 +97,18 @@ def __createEnvVariables(self):
         self.frameEnv["CUE_GPU_MEMORY"] = str(self.rqCore.machine.getGpuMemoryFree())
         self.frameEnv["SP_NOMYCSHRC"] = "1"
 
-        for key in self.runFrame.environment:
-            self.frameEnv[key] = self.runFrame.environment[key]
+        if platform.system() in ("Linux", "Darwin"):
+            self.frameEnv["MAIL"] = "/usr/mail/%s" % self.runFrame.user_name
+            self.frameEnv["HOME"] = "/net/homedirs/%s" % self.runFrame.user_name
+        elif platform.system() == "Windows":
+            for variable in ["SYSTEMROOT", "APPDATA", "TMP", "COMMONPROGRAMFILES"]:
+                self.frameEnv[variable] = os.environ[variable]
```

Comment:
I think this will raise an exception if `os.environ` doesn't contain `variable`. Maybe add a check for `if variable in os.environ` before doing the assignment?

Reply:
Indeed.
These variables are probably always there by default Windows but I'm not 100% sure. So, I will add the check.